getnacelle/nacelle-js
# feat: GraphQL Config and Fragment Masking

## Why are these changes introduced?

Supports [ENG-8700](https://nacelle.atlassian.net/browse/ENG-8700) 



> Example project demonstrating typed fields in content queries

## What is this pull request doing?



This PR supports #408 by adding:
- [GraphQL Config](https://the-guild.dev/graphql/config) support
- [Fragment Masking](https://the-guild.dev/graphql/codegen/plugins/presets/preset-client#fragment-masking) to embrace `@graphql-codegen/client-preset` conventions

## How to Test



1. Check out the `ENG-8700/graphql-config` branch
2. From the monorepo root, `npm i`
3. Navigate to `examples/typedfields`
4. If you haven't already, create a `examples/typedfields/.env.local` with the contents of `examples/typedfields/.env.example`
5. `npm run codegen` - code generation should happen with no errors
6. In any of the `graphql()` calls in `app/`, `CMD` + `space` should display available fields to query, like so:

https://github.com/getnacelle/nacelle-js/assets/5732000/81a3f147-bd60-4fc9-bbef-e7b00c979b91

7. `npm run dev` - all of the pages should continue to work as they do in #408:

https://github.com/getnacelle/nacelle-js/assets/5732000/cc7b881b-4e00-4263-929b-91090f6a015d

8. Check that the updated [README](https://github.com/getnacelle/nacelle-js/blob/ENG-8700/graphql-config/examples/typedfields/README.md) makes sense :)

## Checklist


[ENG-8700]: https://nacelle.atlassian.net/browse/ENG-8700?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
